front-end/src/components/AnalysisPage.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import store from "../store.js";
+import { openMainPage } from "../store/switchPageSlice.js";
+import BaseButton from "./BaseButton.jsx";
+
+class AnalysisPage extends Component {
+
+    openMainPage = () => {
+      store.dispatch(openMainPage());
+    }
+
+    render() {
+        return (
+            <div className="AnalysisPage bg-purple-300 flex" data-testid="analysis-page">
+              <div className="flex gap-y-10 w-full flex-col">
+                <div className="justify-center flex">
+                  <BaseButton
+                    click={this.openMainPage}
+                    text="Return to Main Page" />
+                </div>
+              </div>
+            </div>
+          );
+    }
+}
+
+export default AnalysisPage;
```

Comment:
Awesome addition to this ticket, but if you are going to add it, can you add testing as well?

Reply:
Sure! Would there be any specific features to test right now? The only action you can perform on this page is click on Return to Main Page which is tested in App. Should there be explicit testing for this button?

Reply:
The main test is to make sure that upon the button click, store.dispatch is called with argument openMainPage()